File: problem10.rb
max = 2000000
nums = (2..max).to_a

nums.each do |i|
  if (i > Math.sqrt(max))
    break
  end

  nums.select! { |j| j == i || j % i != 0 }
end

puts nums.reduce { |sum, i| sum + i }

File: javascript/problem10.js
var _ = require( 'lodash' );

var nums = _.range( 3, 2000000, 2 );

var i = 0;
var prime = nums[ i ];
while ( prime < Math.sqrt( 2000000 ) ) {
  nums = nums.filter( n => n == prime || n % prime != 0 );
  prime = nums[ ++i ];
}

nums.unshift( 2 );

var sum = nums.reduce( (sum, n) => sum + n );

console.log( sum );
